### PR TITLE
Convert Throw Expressions to multi-line lambda function

### LINKED ISSUE #97: C# -> VB: Convert Throw Expressions to multi-line lambda function
Converting a throw expression to a multi-line lambda function removes the need for a inline helper method.

### Input code
```C#
static class TestClass
{
    private static object TypeSwitch(this object obj, Func<string, object> matchFunc1, Func<int, object> matchFunc2, Func<object, object> defaultFunc)
    {
        return null;
    }

    private static object ConvertInt(int node)
    {
        return node;
    }

    private static object ConvertString(string node)
    {
        return node;
    }

    public static object Convert(object node)
    {
        return node.TypeSwitch(ConvertString, ConvertInt, _ => throw new NotImplementedException($"Conversion for '{node.GetType()}' not implemented"));
    }
}
```

### Current Conversion
```vbnet
Imports System.Runtime.CompilerServices

Module TestClass
  <Extension()>
  Private Function TypeSwitch(ByVal obj As Object, ByVal matchFunc1 As Func(Of String, Object), ByVal matchFunc2 As Func(Of Integer, Object), ByVal defaultFunc As Func(Of Object, Object)) As Object
    Return Nothing
  End Function

  Private Function ConvertInt(ByVal node As Integer) As Object
    Return node
  End Function

  Private Function ConvertString(ByVal node As String) As Object
    Return node
  End Function

  Function Convert(ByVal node As Object) As Object
    Return node.TypeSwitch(New Func(Of String, Object)(AddressOf ConvertString), New Func(Of Integer, Object)(AddressOf ConvertInt), Function(__) CSharpImpl.__Throw(Of Object)(New NotImplementedException($"Conversion for '{node.[GetType]()}' not implemented")))
  End Function

  Private Class CSharpImpl
    <Obsolete("Please refactor calling code to use normal throw statements")>
    Shared Function __Throw(Of T)(ByVal e As Exception) As T
      Throw e
    End Function
  End Class
End Module
```

### Suggested output
```vbnet
Imports System.Runtime.CompilerServices

Module TestClass
  <Extension()>
  Private Function TypeSwitch(ByVal obj As Object, ByVal matchFunc1 As Func(Of String, Object), ByVal matchFunc2 As Func(Of Integer, Object), ByVal defaultFunc As Func(Of Object, Object)) As Object
    Return Nothing
  End Function

  Private Function ConvertInt(ByVal node As Integer) As Object
    Return node
  End Function

  Private Function ConvertString(ByVal node As String) As Object
    Return node
  End Function

  Function Convert(ByVal node As Object) As Object
    Return node.TypeSwitch(New Func(Of String, Object)(AddressOf ConvertString), New Func(Of Integer, Object)(AddressOf ConvertInt), Function(__)
                                                                                                                                        Throw New NotImplementedException($"Conversion for '{node.[GetType]()}' not implemented")
                                                                                                                                      End Function)
  End Function
End Module
```

### Details
The benefit to not using inline helper methods is it removes the need for the user to manually convert code.

### PR DESCRIPTION
**Pull request by RussellSmith2**
Fixes #97

### CODE REVIEW

File: ICSharpCode.CodeConverter/VB/CommonConversions.cs [hunk 1]
```diff
@@ -389,12 +397,35 @@ public static SyntaxToken ConvertIdentifier(SyntaxToken id)
...
+                             CSSyntaxKind.EnumMemberDeclaration,
+                             CSSyntaxKind.EventDeclaration,
+                             CSSyntaxKind.EventFieldDeclaration,
+                             CSSyntaxKind.FieldDeclaration,
+                             CSSyntaxKind.InterfaceDeclaration,
+                             CSSyntaxKind.MethodDeclaration,
+                             CSSyntaxKind.PropertyDeclaration,
+                             CSSyntaxKind.NamespaceDeclaration,
+                             CSSyntaxKind.StructDeclaration,
+                             CSSyntaxKind.VariableDeclaration);
```
Comment (GrahamTheCoder) [inline ICSharpCode.CodeConverter/VB/CommonConversions.cs]: Is there a link where you got this list or did you figure it out one by one? Should `DelegateDeclaration` be in here too for example?

Comment (GrahamTheCoder) [reply ICSharpCode.CodeConverter/VB/CommonConversions.cs]: I'm tempted to just add all the ones ending in declaration - but then I wonder if there's a parent syntax type that might be easier to test against

```csharp
return id.IsKind(
CSSyntaxKind.VariableDeclaration,
CSSyntaxKind.CatchDeclaration,
CSSyntaxKind.NamespaceDeclaration,
CSSyntaxKind.ClassDeclaration,
CSSyntaxKind.StructDeclaration,
CSSyntaxKind.InterfaceDeclaration,
CSSyntaxKind.EnumDeclaration,
CSSyntaxKind.DelegateDeclaration,
CSSyntaxKind.EnumMemberDeclaration,
CSSyntaxKind.FieldDeclaration,
CSSyntaxKind.EventFieldDeclaration,
CSSyntaxKind.MethodDeclaration,
CSSyntaxKind.OperatorDeclaration,
CSSyntaxKind.ConversionOperatorDeclaration,
CSSyntaxKind.ConstructorDeclaration,
CSSyntaxKind.DestructorDeclaration,
CSSyntaxKind.PropertyDeclaration,
CSSyntaxKind.EventDeclaration,
CSSyntaxKind.IndexerDeclaration,
CSSyntaxKind.GetAccessorDeclaration,
CSSyntaxKind.SetAccessorDeclaration,
CSSyntaxKind.AddAccessorDeclaration,
CSSyntaxKind.RemoveAccessorDeclaration,
CSSyntaxKind.UnknownAccessorDeclaration
);
```

**Comment by GrahamTheCoder** [comment]
Sounds good to me. I've merged it with the `DelegateDeclaration` added

**Comment by RussellSmith2** [comment]
#99 was fixed by my second commit. It is currently still open

**Pull request merged**